pkg/database/mongoDB.go

```diff
@@ -34,19 +35,21 @@ func NewMongoProducts() ProductDB {
 
 func (mp *MongoProducts) Connect() error {
 	uri := mongodbURI()
-	
+
 	// Setting client options
-	clientOptions := options.Client().ApplyURI(uri)
+	opts := options.Client()
+	clientOptions := opts.ApplyURI(uri)
+	opts.Monitor = otelmongo.NewMonitor()
 
 	// Connect to MongoDB
-	client, err := mongo.Connect(context.TODO(), clientOptions)
+	client, err := mongo.Connect(context.Background(), clientOptions)
```

Comment:
See this comment : https://github.com/Ubivius/microservice-template/pull/62/files#r725752203